lib/sse.js

```diff
@@ -125,65 +127,73 @@ SSE.prototype.handler = function (req, res, next) {
-            var resourceNames = _.map(routeNames, function(routeName) {
-                var pluralName = (that.options.inflect) ? inflect.pluralize(routeName) : routeName;
-                return new RegExp(pluralName, 'i');
-            });
-
-            var matchesEitherResource = _.some(resourceNames, function(resourceName) {
-                return resourceName.test(chunk.ns);
-            });
-
-            if (matchesEitherResource) {
-                var id = chunk.ts.getHighBits() + '_' + chunk.ts.getLowBits();
-                var eventName = that.getEventName(routeNames, chunk);
-                var data = that.getData(routeNames[0], chunk);
-
-                 var filters = that.getFilters(req);
-
-                 var passedFilter = _.reduce(filters, function(obj, filter) {
-                     return _.filter([data], _.matches(filter));
-                 }, true);
-
-                //if we have filters, make sure they are passed
-                if (passedFilter.length > 0 || filters.length === 0) {
-                    tinySSE.send({id: id, event: eventName, data: data})(req, res);
+            var collection = that.db.collection('oplog.rs')
+            
+            var lastTS;
+            
+            var consumerFunc = function(err, chunk) {
+                
+                if (!chunk) return;
+                var resourceNames = _.map(routeNames, function(routeName) {
+                    var pluralName = (that.options.inflect) ? inflect.pluralize(routeName) : routeName;
+                    return new RegExp(pluralName, 'i');
+                });
+                
+                var matchesEitherResource = _.some(resourceNames, function(resourceName) {
+                    return resourceName.test(chunk.ns);
+                });
+        
+                if (matchesEitherResource) {
+                    var id = chunk.ts.getHighBits() + '_' + chunk.ts.getLowBits();
+                    var eventName = that.getEventName(routeNames, chunk);
+                    var data = that.getData(routeNames[0], chunk);
+    
+                    var filters = that.getFilters(req);
+    
+                    var passedFilter = _.reduce(filters, function(obj, filter) {
+                        return _.filter([data], _.matches(filter));
+                    }, true);
+    
+                    //if we have filters, make sure they are passed
+                    if (passedFilter.length > 0 || filters.length === 0) {
+                        tinySSE.send({id: id, event: eventName, data: data})(req, res);
+                    }
+   
                 }
-
-                next();
+                
+                lastTS = chunk.ts;
             }
-        });
-
-        docStream.through(consume).errors(function(err) {
-            that.handleError(err, res, docStream);
-        });
+            
+            interval = setInterval(function() {
+                if(lastTS) {
```

Comment:
shouldn't this call itself once it's done processing all of the oplog entries?
as far as I understand the way it's set up now it will call the collection.find after 2s, irregardless of whether the previous action has completed